lib/widgets/app_bar_widget.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:flutter/material.dart';
+import 'package:tech_byte/utils/colors.dart';
+import 'package:tech_byte/utils/constants.dart';
+
+class TBAppBar extends StatelessWidget implements PreferredSizeWidget {
+  final String? title;
+  final List<Widget>? actions;
+  @override
+  final Size preferredSize;
+  const TBAppBar({super.key, this.title, this.actions})
+      : preferredSize = const Size.fromHeight(60);
+
+  @override
+  Widget build(BuildContext context) {
+    return AppBar(
+      backgroundColor: Colors.white,
+      centerTitle: false,
+      titleSpacing: Navigator.of(context).canPop() ? -5 : null,
+      leading: Navigator.of(context).canPop()
+          ? IconButton(
+              onPressed: () => Navigator.of(context).pop(),
+              icon: Image.asset(
+                "assets/chevron_left_icon.png",
+                height: TBDimensions.appBar.iconSize,
+                width: TBDimensions.appBar.iconSize,
+              ))
+          : null,
+      shape: RoundedRectangleBorder(
+        borderRadius: BorderRadius.only(
+            bottomLeft: Radius.circular(TBDimensions.appBar.borderRadius),
+            bottomRight: Radius.circular(TBDimensions.appBar.borderRadius)),
+      ),
+      title: title == null
```

Comment:
Do not like the logic, in both types of AppBars you should provide title, it is just two types of AppBars. This logic is not generic it is too much specific for this case